src/main/groovy/org/pih/warehouse/inventory/QuantityByBinLocationDto.groovy

```diff
@@ -0,0 +1,31 @@
+package org.pih.warehouse.inventory
+
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.product.Category
+import org.pih.warehouse.product.Product
+
+class QuantityByBinLocationDto implements Comparable<QuantityByBinLocationDto> {
+    String id
+
+    String status
+
+    String value
+
+    Category category
+
+    Product product
+
+    InventoryItem inventoryItem
+
+    Location binLocation
+
+    Integer quantity
+
+    Boolean isOnHold
+
+    @Override
+    int compareTo(QuantityByBinLocationDto obj) {
+        inventoryItem?.expirationDate <=> obj?.inventoryItem?.expirationDate
+                ?: binLocation?.name <=> obj?.binLocation?.name
+    }
```

Comment:
I don't mind adding it, but it was only a copy paste from already existing (and working) code and I did not pay much attention to that.

Reply:
Yeah it's a good point @drodzewicz . Whether it should be included in the PR is negotiable since (as @kchelstowski mentioned) this is just a copy-paste from the previous sort implementation. 

If it's not included, then we should make sure we prioritize the following ticket and maybe a spike to test out the different implementations.
https://pihemr.atlassian.net/browse/OBGM-341